lti_consumer/lti.py

```diff
@@ -0,0 +1,281 @@
+"""
+This module encapsulates code which implements the LTI specification.
+
+For more details see:
+https://www.imsglobal.org/activity/learning-tools-interoperability
+"""
+
+import logging
+import urllib
+import json
+
+from .exceptions import LtiError
+from .oauth import get_oauth_request_signature, verify_oauth_body_signature
+
+
+log = logging.getLogger(__name__)
+
+
+def parse_result_json(json_str):
+    """
+    Helper method for verifying LTI 2.0 JSON object contained in the body of the request.
+
+    The json_str must be loadable.  It can either be an dict (object) or an array whose first element is an dict,
+    in which case that first dict is considered.
+    The dict must have the "@type" key with value equal to "Result",
+    "resultScore" key with value equal to a number [0, 1],
+    The "@context" key must be present, but we don't do anything with it.  And the "comment" key may be
+    present, in which case it must be a string.
+
+    Arguments:
+        json_str (unicode):  The body of the LTI 2.0 results service request, which is a JSON string
+
+    Returns:
+        (float, str):  (score, [optional]comment) if verification checks out
+
+    Raises:
+        LtiError: if verification fails
+    """
+    try:
+        json_obj = json.loads(json_str)
+    except (ValueError, TypeError):
+        msg = "Supplied JSON string in request body could not be decoded: {}".format(json_str)
+        log.info("[LTI] %s", msg)
+        raise LtiError(msg)
+
+    # the standard supports a list of objects, who knows why. It must contain at least 1 element, and the
+    # first element must be a dict
+    if not isinstance(json_obj, dict):
+        if isinstance(json_obj, list) and len(json_obj) >= 1 and isinstance(json_obj[0], dict):
+            json_obj = json_obj[0]
+        else:
```

Comment:
this was a speed bump while I was reading through the code. Perhaps something like this:

 if isinstance(json_obj, list) and len(json_obj) >= 1
    json_obj = json_obj[0]

if not isinstance(json_obj, dict):
    #error stuff here
    
A little more clear, covers the same cases and only checks if the object is a dict once.